### PR TITLE
EQL: add the wildcard field type to the IT tests

### PR DESCRIPTION
**Pull request by astefan**
A randomized choice between `wildcard` and `keyword` has been added to IT tests by dynamically changing the mapping of the tested index.

### CODE REVIEW

File: x-pack/plugin/eql/qa/common/src/main/resources/mapping-default.json [hunk 1]
```diff
@@ -72,8 +74,8 @@
         "file_name" : {
             "type" : "text",
             "fields" : {
-                "keyword" : {
-                    "type" : "keyword",
+                "[keyword_like_type]" : {
+                    "type" : "[keyword_like_type]",
                     "ignore_above" : 256
```
Comment (costin) [inline x-pack/plugin/eql/qa/common/src/main/resources/mapping-default.json:79]: Does this setting make sense for wildcard?

File: x-pack/plugin/eql/qa/common/src/main/java/org/elasticsearch/test/eql/DataLoader.java [hunk 2]
```diff
@@ -37,9 +43,19 @@
...
+    private static final Map<String, String[]> replacementPatterns = Collections.unmodifiableMap(getReplacementPatterns());
     static final String indexPrefix = "endgame";
     public static final String testIndexName = indexPrefix + "-1.4.0";
 
+    private static final long FILETIME_EPOCH_DIFF = 11644473600000L;
+    private static final long FILETIME_ONE_MILLISECOND = 10 * 1000;
+
+    private static Map<String, String[]> getReplacementPatterns() {
+        final Map<String, String[]> map = new HashMap<>(1);
+        map.put("[keyword_like_type]", new String[] {"keyword", "wildcard"});
```
Comment (costin) [inline x-pack/plugin/eql/qa/common/src/main/java/org/elasticsearch/test/eql/DataLoader.java:55]: I would use a more descriptive name such as `random_string_type` or `runtime_random_string_type`.

**Review by costin — APPROVED** [review]
LGTM, left one comment mainly on cosmetics, that's all.

File: x-pack/plugin/eql/qa/common/src/main/resources/mapping-default.json (hunk 1 above)
Comment (astefan) [reply x-pack/plugin/eql/qa/common/src/main/resources/mapping-default.json:79]: Yes, it's [the only supported setting of a `wildcard` field type](https://www.elastic.co/guide/en/elasticsearch/reference/master/keyword.html#wildcard-field-type).

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-windows

**Review by matriv — APPROVED** [review]
LGTM

**Pull request merged**